BLSQ/iaso
# IA-3026: In the org unit search, the loading of the map tab should only start when we switch to the map tab

Currently, if you go to the org unit list, and click search, you’ll see in the requests that it loads tiles and up to 3000 org units to display on the map, while it should only load the org units needed for the list. 

These requests should only start when the map is visible on screen. 

Related JIRA tickets : IA-3026

## Self proofreading checklist


## Changes

- refactor the use of react query 
- only fetch map data if on map tab
- fix a bug while deleting a tab (deleting a non selected tab is leading to an inconsistent behaviour)
- fix fit to bound while changing results


## How to test

Play with the org units search and multi tab. 
Make sure it makes only call for the map while on the map.
Order, pagination, filters, try to break it please

## Print screen / video

Nothing to show here as everything should be the same but better

## Notes

⚠️ There is a blsq-component PR [here](https://github.com/BLSQ/bluesquare-components/pull/169) to use with it

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.
